### PR TITLE
add --set-perms to pysogs, let --add-room also use them

### PR DESCRIPTION
**Pull request by tewinget**

### CODE REVIEW

File: sogs/__main__.py [hunk 1]
```diff
@@ -51,6 +57,27 @@
...
+actions.add_argument('--set-perms', help="Sets permissions for the room given by --rooms")
+ap.add_argument(
+    '--users',
+    help="Users to set permissions for with --set-perms; if omitted, set global perms",
+    nargs='+',
+    metavar='SESSIONID',
+)
+ap.add_argument(
+    "--add-perms",
+    help="With --add-room or --set-perms, set these permissions to true; --add-perms=[r][w][u][a]",
```
Comment (jagerman) [inline sogs/__main__.py]:
```suggestion
    help="With --add-room or --set-perms, add these permissions to; should be a string of 1-4 of the letters \"rwua\" for [r]ead, [w]rite, [u]pload, and [a]ccess.",
```

File: sogs/__main__.py [hunk 2]
```diff
@@ -51,6 +57,27 @@
     metavar='SESSIONID',
     help="Delete the the given Session ID(s) as moderator and admins of the room given by --rooms",
 )
+actions.add_argument('--set-perms', help="Sets permissions for the room given by --rooms")
+ap.add_argument(
+    '--users',
+    help="Users to set permissions for with --set-perms; if omitted, set global perms",
```
Comment (jagerman) [inline sogs/__main__.py]:
```suggestion
    help="Users to set permissions for with --set-perms; if omitted, set the default permissions for the room.",
```

File: sogs/__main__.py [hunk 3]
```diff
@@ -51,6 +57,27 @@
...
+    metavar='SESSIONID',
+)
+ap.add_argument(
+    "--add-perms",
+    help="With --add-room or --set-perms, set these permissions to true; --add-perms=[r][w][u][a]",
+)
+ap.add_argument(
+    "--remove-perms",
+    help="With --add-room or --set-perms, set these permissions to false; "
+    "--remove-perms=[r][w][u][a]",
```
Comment (jagerman) [inline sogs/__main__.py]:
```suggestion
    help="With --add-room or --set-perms, deny these permissions; should be a string of 1-4 of the letters \"rwua\" for [r]ead, [w]rite, [u]load, and [a]ccess.",
```

File: sogs/__main__.py [hunk 4]
```diff
@@ -51,6 +57,27 @@
...
+)
+ap.add_argument(
+    "--clear-perms",
+    help="With --add-room or --set-perms, clear room or user overrides on these permissions; "
+    "--clear-perms=[r][w][u][a]",
```
Comment (jagerman) [inline sogs/__main__.py]:
```suggestion
    help="With --add-room or --set-perms, clear room or user overrides on these permissions; should be a string of 1-4 of the letters \"rwua\".",
```

File: sogs/__main__.py [hunk 5]
```diff
@@ -51,6 +57,27 @@
...
 )
+actions.add_argument('--set-perms', help="Sets permissions for the room given by --rooms")
+ap.add_argument(
+    '--users',
+    help="Users to set permissions for with --set-perms; if omitted, set global perms",
+    nargs='+',
+    metavar='SESSIONID',
+)
+ap.add_argument(
+    "--add-perms",
```
Comment (jagerman) [inline sogs/__main__.py:74]: Thoughts on calling this "--grant-perms" instead of "--add-perms"?  (And likewise revoke instead of remove)?

Comment (tewinget) [reply sogs/__main__.py:74]: utter indifference

**Pull request merged**